File: 参考/comp9041-Q/cs9041/test6/equi_filter.pl
#!/usr/bin/perl -w
sub check{
  %data=();
  my @word = @_;
  foreach my $c (@word){
    $test = $c;
    $data{$c}++;
  }
  foreach my $value (values %data){
    if($value != $data{$test}){
      return 0;
    }
  }
  return 1;
}


while($line=<>){
    chomp $line;
    @words = split / +/,$line;
    @answer = ();
    foreach $word (@words){
      if($word){
        %data = ();
        $ori = $word;
        $word =~ tr /[A-Z]/[a-z]/;
        @word = split //,$word;
        if (check(@word)){
          push @answer,$ori;
        }
      }
    }
    print join(" ",@answer),"\n";
}



















# while ($line = <>){
#   @arr = split " ",$line;
#   @answer = ();
#   foreach $x (@arr){
#     $flag = 0;
#     %data = ();
#     $last = $x;
#     @word = ();
#     # $x =~ s/[^a-zA-Z]//g;
#     $x =~ tr/[a-z]/[A-Z]/;
#     @word = split "",$x;
#     # print join"|",@word;
#     foreach $char (@word){
#       $data{$char}++;
#     }
#     # print %data;
#     @count = values %data;
#     $end = $#count;
#     foreach $i (1..$end){
#         if ($count[$i] != $count[$i - 1]){
#           $flag = 1;
#         }
#     }
#     # print $flag;
#     unless ($flag) {
#       push @answer,$last;
#     }
#   }
#   print join " ",@answer;
#   @arr= ();
#   print "\n";
#
# }

File: others/comp9041-Q/cs9041/test6/equi_filter.py
#!/usr/bin/python3
import sys,re
def check(word):
    word = word.lower()
    for c in word:
        if word.count(c) != word.count(word[0]):
            return False
    return True

while True:
    line = sys.stdin.readline()
    if not line: break
    words = re.split("\s+",line.strip())
    answer = [word for word in words if check(word)]
    print(" ".join(answer))



















# import sys
# from collections import defaultdict
# for line in sys.stdin:
#     answer = []
#     l = line.split()
#     for word in l:
#         flag = 1
#         count = defaultdict(list)
#         for x in word:
#             count[x.lower()].append(1)
#         value = list(count.values())
#         for i in range(1,len(value)):
#             if value[i] != value[i-1]:
#                 flag = 0
#         if flag:
#             answer.append(word)
#     # print(answer)
#     print (" ".join(answer))
